fl4health/clients/dynamic_weight_exchange_client.py

```diff
@@ -0,0 +1,121 @@
+        Dynamic weight exchange client used to exchange a dynamic subset of layers per client.
+
+        Args:
+            data_path (Path): path to the data to be used to load the data for client-side training
+            metrics (Sequence[Metric]): Metrics to be computed based on the labels and predictions of the client model
+            device (torch.device): Device indicator for where to send the model, batches, labels etc. Often 'cpu' or
+                'cuda'
+            loss_meter_type (LossMeterType, optional): Type of meter used to track and compute the losses over
+                each batch. Defaults to LossMeterType.AVERAGE.
+            metric_meter_type (MetricMeterType, optional): Type of meter used to track and compute the metrics over
+                each batch. Defaults to MetricMeterType.AVERAGE.
+            checkpointer (Optional[TorchCheckpointer], optional): Checkpointer to be used for client-side
+                checkpointing. Defaults to None.
+        """
+        super().__init__(
+            data_path=data_path,
+            metrics=metrics,
+            device=device,
+            loss_meter_type=loss_meter_type,
+            metric_meter_type=metric_meter_type,
+        )
+        # Initial model parameters to be used in calculating weight shifts during training
+        self.initial_model: nn.Module
+        # Parameter exchanger to be used in server-client exchange of dynamic layers.
+        self.parameter_exchanger: NormDriftParameterExchanger
+
+    def setup_client(self, config: Config) -> None:
+        """
+        Setup the components of the client neccessary for client side training and parameter exchange. Mostly handled
+        by a call to the basic client flow, but also sets up the initial model to facilitate storage of initial
+        parameters during training
+
+        Args:
+            config (Config): Configuration used to setup the client properly
+        """
+        super().setup_client(config)
+        self.initial_model = copy.deepcopy(self.model).to(self.device)
+
+    def get_parameter_exchanger(self, config: Config) -> ParameterExchanger:
+        """
+        This method configures and instantiates a NormDriftParameterExchanger to be used in dynamic weight exchange.
+
+        Args:
+            config (Config): Configuration used to setup the weight exchanger properties for dynamic exchange
+
+        Returns:
+            ParameterExchanger: This exchanger handles the exchange orchestration between clients and server during
+                federated training
+        """
+        normalize = self.narrow_config_type(config, "normalize", bool)
+        filter_by_percentage = self.narrow_config_type(config, "filter_by_percentage", bool)
+        norm_threshold = self.narrow_config_type(config, "norm_threshold", float)
+        exchange_percentage = self.narrow_config_type(config, "exchange_percentage", float)
+        parameter_exchanger = NormDriftParameterExchanger(
```

Comment:
We may also want the user to be able to configure what norm we use but default to the 2-norm since it is probably what people are looking to use most of the time.

Reply:
Yup definitely!